### PR TITLE
Runtime modules into separated crates

### PR DESCRIPTION
**Pull request by fmiguelgarcia**
It will improve our compilation time.

- [x] Common module crate:
    - [x] Remove circular dependency `Identity` & `Balances`.
    - [x] Common module tools and types. 
- [x] Move out `Identity` module into a separated crate.
- [x] Move out `Balances` module into a separated crate.
- [x] Keep other modules under `Runtime`. They will be split later.

And more features like
 - [x] Ugly `ElevateTrait` trick has been removed.
 - [x] New folder structure.

### CODE REVIEW

**Comment by maxsam4** [comment]
List of changes that are required:
- Fix directory structure in runtime/runtime/cargo.toml
- Add `clear_on_drop = { version = "0.2.3", features = ["no_cc"] }` to runtime/cargo.toml

 I will keep updating this as I make changes. Not committing them right away bcz I might have to roll back some. Also, I don't think I'll be able to do all the fixes so treat these as just some helpers.

File: pallets/common/src/currency.rs [hunk 1]
```diff
@@ -0,0 +1,250 @@
+use crate::traits::{BalanceLock, CommonTrait};
```
Comment (fmiguelgarcia) [inline pallets/common/src/currency.rs]: I need to remove this file completely

File: .circleci/config.yml [hunk 2]
```diff
@@ -34,8 +34,8 @@ jobs:
           paths:
             - "~/.cargo"
             - "./target"
-            - "./runtime/target"
-            - "./runtime/wasm/target"
+            - "./pallets/runtime/target"
```
[the remark this answers is not in the document]
Comment (fmiguelgarcia) [reply .circleci/config.yml]: Yes, that was the initial idea, but it seems it requires more work to decouple runtime from other modules.
We could do that in another ticket.

**Comment by satyamakgec** [comment]
thanks, @fmiguelgarcia for doing all the hard work this is a big chunk of work.
It is very hard to review that PR. I did look over through it and the changes seem reasonable, So I recommend merging by assuming everything will run fine. But I have a couple of questions before the merge.

1-  Is the module index will remain the same as we rip out the events from some of the sub-modules?
2- What is the motivation of not converting the other modules as the pallet?
3- How the storage prefixes are working in the trait/balances.rs?

**Pull request merged**